udata_front/theme/gouvfr/assets/js/components/AdminTable/AdminDatasetsTable/AdminDatasetsTable.vue

```diff
@@ -0,0 +1,191 @@
+      </tr>
+    </thead>
+    <tbody>
+      <tr v-for="dataset in datasets" :key="dataset.id">
+        <AdminTableTd>
+          <AdminContentWithTooltip>
+            <a class="fr-link" :href="getDatasetLinkToAdmin(dataset)">
+              {{ dataset.title }}
+            </a>
+          </AdminContentWithTooltip>
+        </AdminTableTd>
+        <AdminTableTd>
+        <AdminBadge :type="dataset.private ? 'default' : 'info'">{{ dataset.private ? t('Private') : t('Public') }}</AdminBadge>
+        </AdminTableTd>
+        <AdminTableTd>
+          {{ formatDate(dataset.created_at) }}
+        </AdminTableTd>
+        <AdminTableTd>
+          {{ formatDate(dataset.last_modified) }}
+        </AdminTableTd>
+        <AdminTableTd>
+          {{ getFilesCount(dataset) }}
+        </AdminTableTd>
+        <AdminTableTd>
+          <QualityScoreTooltip :dataset="dataset"/>
+        </AdminTableTd>
+        <AdminTableTd class="monospace">
+          {{ summarize(dataset.metrics.discussions) }}
+        </AdminTableTd>
+        <AdminTableTd class="monospace">
+          {{ summarize(dataset.metrics.views) }}
+        </AdminTableTd>
+        <AdminTableTd class="monospace">
+          {{ summarize(getMetrics(dataset)) }}
+        </AdminTableTd>
+        <AdminTableTd class="monospace">
+          {{ summarize(dataset.metrics.reuses) }}
+        </AdminTableTd>
+        <AdminTableTd class="monospace">
+          {{ summarize(dataset.metrics.followers) }}
+        </AdminTableTd>
+      </tr>
+    </tbody>
+  </AdminTable>
+</template>
+<script setup lang="ts">
+import { formatDate } from "@etalab/data.gouv.fr-components";
+import type { Dataset, DatasetV2 } from '@etalab/data.gouv.fr-components';
+import { ref } from 'vue';
+import { useI18n } from 'vue-i18n';
+import AdminBadge from "../../AdminBadge/AdminBadge.vue";
+import AdminContentWithTooltip from "../../AdminContentWithTooltip/AdminContentWithTooltip.vue";
+import AdminTable from "../Table/AdminTable.vue";
+import AdminTableTd from "../Table/AdminTableTd.vue";
+import AdminTableTh from "../Table/AdminTableTh.vue";
+import Tooltip from '../../Tooltip/Tooltip.vue';
+import QualityScoreTooltip from "../../dataset/QualityScore/QualityScoreTooltip/QualityScoreTooltip.vue";
+import { metrics_api } from '../../../plugins/api';
+import { admin_root } from '../../../config';
+import { summarize } from '../../../helpers';
+import { DatasetSortedBy, SortDirection } from '../../../types';
+
+const emit = defineEmits<{
+  (event: 'sort', sort: string): void
+}>();
+
+defineProps<{
+  datasets: Array<Dataset | DatasetV2>
+}>();
+
+const metrics = ref<Record<string, number>>({});
+const sort = ref<SortDirection>('desc');
+const sortedBy = ref<DatasetSortedBy>('created');
+
+const { t } = useI18n();
+
+function updateSort(column: DatasetSortedBy, direction: SortDirection) {
+  sort.value = direction;
+  sortedBy.value = column;
+  emit('sort', `${sort.value === 'asc' ? "" : "-"}${sortedBy.value}`);
```

Comment:
Do we have to use signals? I find it hard to follow

Reply:
I think I forgot to reply to your question. 

This is done to follow the "pure presentational" [best practice](https://storybook.js.org/docs/writing-stories/build-pages-with-storybook#pure-presentational-pages) for components or pages of storybook. It moves logic and API request out of individual components to a single wrapper.

Reply:
I'm not a big fan of signals either… But if it's working why not…